plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,134 @@
+        if lfs.attributes(self.cover_image_path ..".bak") then
+            os.execute("mv " .. self.cover_image_path .. ".bak " .. self.cover_image_path)
+        end
+    end
+end
+
+function CoverImage:onDocSettingsLoad(doc_settings, document)
+   logger.dbg("CoverImage: onDocSettingsLoad")
+   if self.enabled then
+        local lfs = require("libs/libkoreader-lfs")
+        local image = document:getCoverPageImage()
+        if image then
+            -- Do not override an existing backup, as this only exists on a crash or abnormal update of KOReader.
+            -- This is very usefull, if an image shall be used as a system screensaver (e.g. on Tolino as /sdcard/suspend_other.jpg)
+            if not lfs.attributes(self.cover_image_path ..".bak") then
+                os.execute("cp " .. self.cover_image_path .. " " .. self.cover_image_path .. ".bak" )
+            end
+            Device.screen.bb.writePNG(image, self.cover_image_path, false)
+        end
+    end
+end
+
+function CoverImage:addToMainMenu(menu_items)
+    menu_items.coverimage = {
+--        sorting_hint = "device", -- maybe this plugin is better situated in device?
+        text_func = function() return _("Cover Image") end,
+        checked_func = function() return self:_enabled() end,
+        sub_item_table = {
+            {
+                text_func = function() return self.cover_image_path and T(_("Cover Image (%1)"), self.cover_image_path)
+                    or _("Cover Image (none)") end,
+                checked_func = function() return self:_enabled() end,
+                callback = function(menu)
+                local InputDialog = require("ui/widget/inputdialog")
+                local sample_input
+                sample_input = InputDialog:new{
+                    title = _("Filename for Cover-Image"),
+                    input = self.cover_image_path,
+                    input_type = "string",
+                    description = _("You can enter the filename of the cover image here."),
+                    buttons = {
+                        {
+                            {
+                                text = _("Cancel"),
+                                callback = function()
+                                    UIManager:close(sample_input)
+                                    menu:updateItems()
+                                end,
+                            },
+                            {
+                                text = _("Disable"),
+                                callback = function()
+                                    self.cover_image_path = sample_input:getInputText()
+                                    G_reader_settings:saveSetting("cover_image_path", self.cover_image_path)
+                                    self.enabled = false
+                                    G_reader_settings:saveSetting("cover_image_enabled", false)
```

Comment:
I haven't seen a screenshot of this widget - but it might be confusing to have a InputDialog with the path and `Cancel|Disable|Enable` buttons.

May be you should split that to 2 menu items:
`Set cover image path` => InputDialog with just `Cancel | Save` (dunno if we should show the path in the menu item)
`[ ] Save cover image on book opening`
`[ ] Restore image on exit`